### ISSUE TITLE
[MU4 Issue] Crash when deleting horizontal frame in scores created with MuseScore 3.x

### ISSUE THREAD

**Issue by samueldy** [musescore/MuseScore#15513]
**Describe the bug**

Posting here after [advised to do so](https://musescore.org/en/node/339710) in the forums.

Opened a score I had made with MuseScore 3.x (I think) and had used a horizontal frame to make a first-system indent. Although the file appeared to convert to MS4 format, attempting to delete this horizontal frame so I could use the first system indent option under Format > Style... resulted in a core dump.

**To Reproduce**
1. Open attached MWE file in MuseScore 4.
2. Click on the horizontal frame to the left of the first measure, or on its handle.
3. Press delete key.
4. Core dumps.

Last few lines of debug output:

```
09:39:23.651 | DEBUG | 139848071509696 | Score      | tick2segment: no measure for tick -1
09:39:26.881 | DEBUG | main_thread | UiActionsRegister | updateEnabledAll: currentCtx: UiCtxNotationFocused
09:39:29.818 | DEBUG | main_thread | ShortcutsController | activate: Del
09:39:29.818 | INFO  | main_thread | ActionsDispatcher | dispatch: try call action: notation-delete
zsh: segmentation fault (core dumped)  mscore -d
```

`UiActionsRegister` line emitted when I clicked the horizontal frame; the rest emitted when I pressed the delete key. Seems to fail no matter whether clicking the frame or the handle on the frame, or whether the frame is before the first (pickup) measure or later on in the score.

Issue does not appear on scores created anew with MuseScore 4.

**Expected behavior**
Pressing the delete key should delete the horizontal frame and not crash program.

**Screenshots**
![image](https://user-images.githubusercontent.com/43612797/209163063-4d6d19ab-874d-4ccc-90d6-8cd476718526.png)

**Platform information**

- Linux manjaro-ws 6.0.15-1-MANJARO `#`1 SMP PREEMPT_DYNAMIC Wed Dec 21 21:14:38 UTC 2022 x86_64 GNU/Linux
- Manjaro Linux 22.0.0
- MuseScore4 4.0.0, package musescore 4.0-2 from community Manjaro testing repo.

**Additional context**

- Minimum working example file attached (in zip because cannot attach .mscz file here)
[mwe.mscz.zip](https://github.com/musescore/MuseScore/files/10287892/mwe.mscz.zip)

**Comment by Jojo-Schmitz** [comment]
Confirmed on Windows 11

**Comment by cbjeukendrup** [comment]
A crash with a similar stack trace came up in https://musescore.org/en/node/343469.

**Comment by DmitryArefiev** [comment]
Fixed in #16181

**Comment by DmitryArefiev** [comment]
Retested on Win10
